modules/apps/27-interchain-accounts/controller/client/cli/tx.go

```diff
@@ -101,18 +104,37 @@ appropriate relative timeoutTimestamp must be provided with flag {relative-packe
 				}
 			}
 
-			relativeTimeoutTimestamp, err := cmd.Flags().GetUint64(flagRelativePacketTimeout)
+			timeoutTimestamp, err := cmd.Flags().GetUint64(flagPacketTimeoutTimestamp)
 			if err != nil {
 				return err
 			}
 
-			msg := types.NewMsgSendTx(owner, connectionID, relativeTimeoutTimestamp, icaMsgData)
+			absoluteTimeouts, err := cmd.Flags().GetBool(flagAbsoluteTimeouts)
+			if err != nil {
+				return err
+			}
+
+			// NOTE: relative timeouts using block height are not supported.
+			if !absoluteTimeouts {
+				// use local clock time as reference time for calculating timeout timestamp.
+				if timeoutTimestamp != 0 {
```

Comment:
Would it be possible to combine these two conditions in a single `if` statement? Like `if !absoluteTimeouts && timeoutTimestamp != 0`.

Reply:
Also, is it possible to have `absoluteTimeouts = true` and `timeoutTimestamp = 0`? In that case we would broadcast a message with a 0 timeout? Would it be better just to return an error early instead?

Reply:
I can see if I can simply the logic here. 

> absoluteTimeouts = true and timeoutTimestamp = 0

This should be possible, but if so then timeHeight needs to be set. Either way it would fail basic validation if neither is set. Maybe the logic should be like:
 `if !absoluteTimeouts && timeoutHeight.IsZero() => err`